crates/extensions/tedge_mqtt_bridge/src/lib.rs

```diff
@@ -377,28 +381,16 @@ async fn half_bridge(
 
             // Keep track of packet IDs so we can acknowledge messages
             Event::Outgoing(Outgoing::Publish(pkid)) => {
-                if pkid == 0 {
-                    // Messages with pkid 0 (meaning QoS=0) should not be added to the hashmap
-                    // as multiple messages with the pkid=0 can be received
-                    match companion_bridge_half.recv().await {
-                        // A message was forwarded by the other bridge half, note the packet id
-                        Some(Some((topic, msg))) => {
-                            loop_breaker.forward_on_topic(topic, &msg);
-                        }
-
-                        // A healthcheck message was published, ignore this packet id
-                        Some(None) => {}
-
-                        // The other bridge half has disconnected, break the loop and shut down the bridge
-                        None => break,
-                    }
-                } else if let hash_map::Entry::Vacant(e) = forward_pkid_to_received_msg.entry(pkid)
-                {
+                if let hash_map::Entry::Vacant(e) = forward_pkid_to_received_msg.entry(pkid) {
                     match companion_bridge_half.recv().await {
                         // A message was forwarded by the other bridge half, note the packet id
                         Some(Some((topic, msg))) => {
                             loop_breaker.forward_on_topic(topic, &msg);
-                            e.insert(msg);
+                            if pkid != 0 {
+                                // Messages with pkid 0 (meaning QoS=0) should not be added to the hashmap
+                                // as multiple messages with the pkid=0 can be received
+                                e.insert(msg);
+                            }
```

Comment:
Can this `pkid = 0` case be covered with a unit test?

Reply:
Not easily. In an ideal world, we would have some deterministic tests where we control the broker we connect to directly, so we can send QoS 0 messages. This would also shore up the existing integration tests, which have to flood the bridge with thousands of messages before disconnecting to reproduce the desired behaviour.

Reply:
> Can this `pkid = 0` case be covered with a unit test?

This is tested by the system test "Support publishing QoS 0 messages to c8y topic #2960".

Sure, this test is not deterministic (as messages with QoS 0 might not be delivered), but I think that is enough.